common/authen.js

```diff
@@ -8,16 +8,40 @@
         // authn API no longer communicates through ermrest, removing the need to check for ermrest location
         var serviceURL = $window.location.origin;
 
+        // Private variable to store current session object
+        var _session = null;
+
+        var _changeCbs = [];
+
         return {
 
             getSession: function() {
                 return $http.get(serviceURL + "/authn/session").then(function(response) {
+                    _session = response.data;
```

Comment:
Shouldn't you call `setSessionValue()` here? This seems like the only place that function would ever be called. If you don't think it should be changed, then maybe `setSessionValue()` doesn't need to be defined.